jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -41,89 +43,120 @@
  * Read an uint8_t value from the given arraybuffer
  */
-static ecma_number_t
+static ecma_value_t
 ecma_typedarray_get_uint8_element (lit_utf8_byte_t *src) /**< the location in the internal arraybuffer */
 {
   uint8_t num = (uint8_t) *src;
-  return (ecma_number_t) num;
+  return ecma_make_number_value (num);
 } /* ecma_typedarray_get_uint8_element */
 
 /**
  * Read an int16_t value from the given arraybuffer
  */
-static ecma_number_t
+static ecma_value_t
 ecma_typedarray_get_int16_element (lit_utf8_byte_t *src) /**< the location in the internal arraybuffer */
 {
   int16_t num;
   memcpy (&num, src, sizeof (int16_t));
-  return (ecma_number_t) num;
+  return ecma_make_number_value (num);
 } /* ecma_typedarray_get_int16_element */
 
 /**
  * Read an uint16_t value from the given arraybuffer
  */
-static ecma_number_t
+static ecma_value_t
 ecma_typedarray_get_uint16_element (lit_utf8_byte_t *src) /**< the location in the internal arraybuffer */
 {
   uint16_t num;
   memcpy (&num, src, sizeof (uint16_t));
-  return (ecma_number_t) num;
+  return ecma_make_number_value (num);
 } /* ecma_typedarray_get_uint16_element */
 
 /**
  * Read an int32_t value from the given arraybuffer
  */
-static ecma_number_t
+static ecma_value_t
 ecma_typedarray_get_int32_element (lit_utf8_byte_t *src) /**< the location in the internal arraybuffer */
 {
   int32_t num;
   memcpy (&num, src, sizeof (int32_t));
-  return (ecma_number_t) num;
+  return ecma_make_number_value (num);
 } /* ecma_typedarray_get_int32_element */
 
 /**
  * Read an uint32_t value from the given arraybuffer
  */
-static ecma_number_t
+static ecma_value_t
 ecma_typedarray_get_uint32_element (lit_utf8_byte_t *src) /**< the location in the internal arraybuffer */
 {
   uint32_t num;
   memcpy (&num, src, sizeof (uint32_t));
-  return (ecma_number_t) num;
+  return ecma_make_number_value (num);
 } /* ecma_typedarray_get_uint32_element */
 
 /**
  * Read a float value from the given arraybuffer
  */
-static ecma_number_t
+static ecma_value_t
 ecma_typedarray_get_float_element (lit_utf8_byte_t *src) /**< the location in the internal arraybuffer */
 {
   float num;
   memcpy (&num, src, sizeof (float));
-  return (ecma_number_t) num;
+  return ecma_make_number_value (num);
 } /* ecma_typedarray_get_float_element */
 
 /**
  * Read a double value from the given arraybuffer
  */
-static ecma_number_t
+static ecma_value_t
 ecma_typedarray_get_double_element (lit_utf8_byte_t *src) /**< the location in the internal arraybuffer */
 {
   double num;
   memcpy (&num, src, sizeof (double));
-  return (ecma_number_t) num;
+  return ecma_make_number_value (num);
 } /* ecma_typedarray_get_double_element */
 
+#if ENABLED (JERRY_BUILTIN_BIGINT)
+/**
+ * Read a bigint64 value from the given arraybuffer
+ */
+static ecma_value_t
+ecma_typedarray_get_bigint64_element (lit_utf8_byte_t *src) /**< the location in the internal arraybuffer */
+{
+  uint64_t digit[1];
+  memcpy (&digit[0], src, sizeof (uint64_t));
```

Comment:
Is unaligned address allowed? Regardless, I think most of the time this is aligned so we can copy it directly which is much faster:
```
if (JERRY_LIKELY ((((uintptr_t) src) & (8 * sizeof (uint64_t) - 1)) == 0))
{
  digit[0] = *(uint64_t*) src;
}
else
{
  memcpy (&digit[0], src, sizeof (uint64_t));
}
```

Reply:
Btw you can create a macro for this, since it needs to be used frequently.